src/docfx/lib/markdown/MarkdigUtility.cs

```diff
@@ -153,6 +154,69 @@ public static string ToPlainText(this MarkdownObject containerBlock)
             return writer.ToString();
         }
 
+        public static bool IsVisible(this MarkdownObject markdownObject)
+        {
+            var visiable = false;
+            Visit(markdownObject, node =>
+            {
+                if (node is HtmlBlock htmlBlock)
+                {
+                    visiable = visiable || VisibleHtml(htmlBlock.Lines.ToString());
+                    return true;
+                }
+
+                if (node is HtmlInline htmlInline)
+                {
+                    visiable = visiable || VisibleHtml(htmlInline.Tag);
+                    return false;
+                }
+
+                if (node is LeafBlock leafBlock &&
+                    (leafBlock.Inline is null || !leafBlock.Inline.Any()))
+                {
+                    visiable = true;
+                    return true;
+                }
+
+                if (node is LeafInline)
+                {
+                    visiable = true;
+                    return true;
+                }
+
+                return false;
+            });
+
+            return visiable;
+
+            bool VisibleHtml(string html)
+            {
+                var reader = new HtmlReader(html);
+                while (reader.Read(out var token))
+                {
+                    // check if there are text
+                    if (token.Type == HtmlTokenType.Text)
+                    {
+                        if (!string.IsNullOrWhiteSpace(token.RawText.ToString()))
```

Comment:
Save the `ToString()` allocation

```suggestion
                        if (!token.RawText.IsWhiteSpace())
```

Reply:
can't find the IsWhiteSpace extension method